pkg/ddl/ingest/backend.go

```diff
@@ -124,9 +136,10 @@ func (bc *litBackendCtx) handleErrorAfterCollectRemoteDuplicateRows(
 				return errors.Trace(tikv.ErrKeyExists)
 			}
 			indexName := tErr.Args()[1]
-			valueStr := tErr.Args()[2]
+			//nolint: forcetypeassert
+			valueStr := tErr.Args()[2].([]string)
 
-			return errors.Trace(tikv.ErrKeyExists.FastGenByArgs(valueStr, indexName))
+			return errors.Trace(tikv.ErrKeyExists.FastGenByArgs(strings.Join(valueStr, "-"), indexName))
```

Comment:
```suggestion
			// same as MySQL, use dash to separate values.
			return errors.Trace(tikv.ErrKeyExists.FastGenByArgs(strings.Join(valueStr, "-"), indexName))
```

Reply:
I'll encapsulate a function for generating `ErrKeyExists`

Reply:
> I'll encapsulate a function for generating ErrKeyExists

I'll do it in a separate PR, because it's related to many components, and there's another bug to be fixed...

And I plan not to add this comment, after the next PR it will be more clear.

Reply:
https://github.com/pingcap/tidb/pull/54382/commits/eb097bed669a55a22a295f8b341e7b3dc0649a01